### PR TITLE
feat: add annotations to band plot

### LINKED ISSUE #1484: Annotations: Support More Graph Types
### Currently Supported:
- [x] Line Graph
- [x] Single Stat With Line

### Acceptance Criteria
- [ ] Band

### PR DESCRIPTION
**Pull request by jrenee42**
Closes #1484

* adds annotations to band plot; with the requisite giraffe upgrade
   * refactored out annotation layer code, simpler to add now
* adds back two skipped e2e tests that pass now

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 58 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=1542)

File: src/visualization/utils/annotationUtils.ts [hunk 1]
```diff
@@ -199,6 +202,49 @@ export const makeAnnotationLayer = (
...
+  cellID: string,
+  xColumn: string,
+  yColumn: string,
+  groupKey: string[],
+  annotations: AnnotationsList,
+  annotationsAreVisible: boolean,
+  dispatch: Dispatch<any>,
+  eventPrefix = 'xyplot'
+) => {
+  if (isFlagEnabled('annotations')) {
```
Comment (hoorayimhelping) [inline src/visualization/utils/annotationUtils.ts:217]: what do you think of inverting this to a guard statement like:

```ts
if (!isFlagEnabled('annotations')) {
  return
}
```

that way, we get the body of the function at the highest indentation level

**Review by hoorayimhelping — COMMENTED** [review]
question about style around the flag check

File: src/visualization/utils/annotationUtils.ts (hunk 1 above)
Comment (jrenee42) [reply src/visualization/utils/annotationUtils.ts:217]: sure, why not.

**Review by hoorayimhelping — APPROVED** [review]

**Pull request merged**